lib/rack/tracker/google_tag_manager/template/google_tag_manager_head.erb

```diff
@@ -6,9 +6,18 @@
 </script>
 <% end %>
 
-<script>(function(w,d,s,l,i){w[l]=w[l]||[];w[l].push({'gtm.start':
+<script>
+<% if options[:turbolinks] %>
+  document.addEventListener('turbolinks:load', function(event) {
+    var url = event.data.url;
+    dataLayer.push({'event':'pageView','virtualUrl': url});
+  });
```

Comment:
sadly I don't know enough about the google tag manager... is google observing the dataLayer array after the page is loaded? is it enough to add entries to the array to get tracked by google?

Reply:
This add a data variable to your GTM that you will need to trigger on it changing and being present.

You can modify your GTM to do add trigger based on History Change (built in variable).  Link this trigger to GA PageView Tag in addition to your all pages trigger.

![image](https://user-images.githubusercontent.com/594229/88345750-834de800-ccfb-11ea-8b9e-226dae5d66ce.png)

Reply:
https://support.google.com/tagmanager/answer/7679322?hl=en#:~:text=Help%20CenterCommunity-,History%20change%20trigger,in%20single%20page%20web%20applications.